test/unit/deadline_job_attachments/bucket_sweeper/test_sweeper_processor.py

```diff
@@ -0,0 +1,240 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+from botocore.exceptions import BotoCoreError
+import pytest
+import os
+import csv
+from typing import Dict, List, Any
+from pathlib import Path
+from unittest.mock import Mock
+from deadline.job_attachments.bucket_sweeper.sweeper_processor import SweeperProcessor
+from deadline.job_attachments.exceptions import (
+    SweeperProcessorError,
+    JobAttachmentS3BotoCoreError,
+)
+
+
+@pytest.fixture
+def mock_clients() -> Dict[str, Mock]:
+    """Fixture to create mock AWS clients"""
+    return {
+        "s3": Mock(),
+        "s3_control": Mock(),
+        "deadline": Mock(),
+        "storage": Mock(),
+        "job_attachments": Mock(),
+    }
+
+
+@pytest.fixture
+def processor(mock_clients: Dict[str, Mock]) -> SweeperProcessor:
+    """Fixture to create SweeperProcessor instance with mock clients"""
+    return SweeperProcessor(
+        s3_client=mock_clients["s3"],
+        s3_control_client=mock_clients["s3_control"],
+        deadline_client=mock_clients["deadline"],
+        farm_id="test-farm",
+        account_id="test-account-id",
+        role_arn="test-role-arn",
+        bucket_name="test-bucket",
+    )
+
+
+@pytest.fixture
+def test_dir(tmp_path: Path) -> Path:
+    """Create and cleanup a test directory."""
+    test_directory: Path = tmp_path / "bucket_sweeper_test"
+    test_directory.mkdir(exist_ok=True)
+
+    return test_directory
+
+
+class TestSweeperProcessor:
+    def test_create_tag_manifest_empty_list(
+        self, processor: SweeperProcessor, test_dir: Path
+    ):
+        """Test creating a tag manifest with an empty delete list."""
+        manifest_path: str = processor._create_tag_manifest(str(test_dir), [])
+
+        with open(manifest_path, "r") as file:
+            assert file.read() == ""
+
+    def test_create_tag_manifest_io_error(
+        self,
+        processor: SweeperProcessor,
+        test_dir: Path,
+        monkeypatch: pytest.MonkeyPatch,
+    ):
+        """Test creating a tag manifest when creation fails."""
+
+        def mock_open(*args, **kwargs):
+            raise IOError("Mocked IO Error")
+
+        monkeypatch.setattr("builtins.open", mock_open)
+
+        with pytest.raises(SweeperProcessorError):
+            processor._create_tag_manifest(str(test_dir), ["object_key"])
```

Comment:
```suggestion
        def mock_open(*args, **kwargs):
            raise IOError("Mocked IO Error")

        monkeypatch.setattr("builtins.open", mock_open)

        with pytest.raises(SweeperProcessorError) as raised_err:
            processor._create_tag_manifest(str(test_dir), ["object_key"])

        assert "Mocked IO Error" in str(raised_err.value)
```

Can we validate that the error was raised due to the mock we setup? And not some other exception?